pkg/descheduler/descheduler.go

```diff
@@ -302,7 +302,7 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
 						continue
 					}
 					evictorFilter := evictions.NewEvictorFilter(nodes, getPodsAssignedToNode, evictLocalStoragePods, evictSystemCriticalPods, ignorePvcPods, evictBarePods, evictions.WithNodeFit(nodeFit), evictions.WithPriorityThreshold(thresholdPriority))
-					f(ctx, rs.Client, strategy, nodes, podEvictor, evictorFilter, getPodsAssignedToNode)
+					f(context.WithValue(ctx, "strategyName", string(name)), rs.Client, strategy, nodes, podEvictor, evictorFilter, getPodsAssignedToNode)
```

Comment:
is this intended to be the final approach or just a work in progress step? (passing values through the context)

Reply:
I think a better approach would be to define an "options" struct to pass as an optional param to `EvictPod`. that struct can have `strategy` and `reasons` fields to start with, and if we decide to add more options in the future then we don't need to change `EvictPod`'s signature. 

what do you think about something like that? As this is, I don't think it's an appropriate use of `context`

Reply:
that said, I don't need to block on this right now since this PR has been open for a while (and that's my bad for not getting around to reviewing until now). If my idea sounds good I'll take it as a follow up refactor and we can unhold this PR

Reply:
We might as well move the strategy name from `EvictPod`. It's used mostly for metrics. Then two lines for logging which can be logged without the strategy name (the same for the reason). Instead, printing both the strategy name and the reason (if there's any) outside of `EvictPod` through a wrapper which will get introduced after the framework primitives are implemented.

My reasoning is to use `EvictPod` only for the actual eviction. The method does not need to know anything about the strategy/plugin/reason/etc. in order to evict a pod. We can have higher invokers to log the additional information.

> If my idea sounds good I'll take it as a follow up refactor and we can unhold this PR

+1 for refactoring the code more in the follow up PR.

Reply:
I think the idea of passing additional info/options to `EvictPod` makes sense. It's already a wrapper for a private `evictPod` function (not sure why though), but as the public interface it seems like a good spot to expose a handle for customization/logging/metrics.